Source/SwiftLintFramework/Rules/Style/FileTypesOrderRule.swift

```diff
@@ -137,19 +137,10 @@ public struct FileTypesOrderRule: ConfigurationProviderRule, OptInRule {
         }
 
         let substructuresSortedByBodyLength = priorityKindSubstructures.sorted { lhs, rhs in
-            return (lhs.bodyLength ?? 0) > (rhs.bodyLength ?? 0)
-        }
-
-        guard let mainTypeSubstructure = substructuresSortedByBodyLength.first else {
-            let substructuresSortedByBodyLength = dict.substructure.sorted { lhs, rhs in
-                return (lhs.bodyLength ?? 0) > (rhs.bodyLength ?? 0)
-            }
-
-            // specify substructure with longest body as main type if existent
-            return substructuresSortedByBodyLength.first
+            return (lhs.bodyLength ?? 0) >= (rhs.bodyLength ?? 0)
```

Comment:
This subtle change (> to >=) ensures that the "main" substructure (struct, enum or class) is assumed to be the first of the longest substructures if more than one is tied for longest. Previously, the last of equal length longest substructures was declared the main type, making fixing the warning by rearranging impossible.